loleaflet/src/layer/tile/CalcTileLayer.js

```diff
@@ -2150,22 +2159,13 @@ L.SpanList = L.Class.extend({
 	},
 
 	_getSpanData: function (spanid) {
-
+		// TODO: Check if data is changed by the callers. If not, return the pointer instead.
+		var clone = {};
 		var span = this._spanlist[spanid];
-		var dataClone = undefined;
-		if (span.data) {
-			dataClone = {};
-			Object.keys(span.data).forEach(function (key) {
-				dataClone[key] = span.data[key];
-			});
-		}
-
-		return {
-			start: spanid ? this._spanlist[spanid - 1].index + 1 : 0,
-			end: span.index,
-			size: span.value,
-			data: dataClone
-		};
+		Object.keys(span).forEach(function (key) {
+			clone[key] = span[key];
```

Comment:
The reason why we needed a clone of the 'data' member was that it is passed outside of L.SpanList and callers could potentially mutate that which is not ideal, but as long as L.SpanList is strongly tied and used by CalcTileLayer, this should be ok. 

Apart for that, I think it would good to do the renaming of index/value etc as currently it doubles the memory usage of a SpanList which might matter for a sheet with lots of spans especially one with auto-filter applied.

Reply:
@dennisfrancis i think you commented on TODO's, thanks for that :) I am planning to create an easy hack out of that extra variables :)